test/HLSL/passes/reassociate/constant-counting.ll

```diff
@@ -0,0 +1,189 @@
+; RUN: opt %s -reassociate -S | FileCheck %s
+
+; Issue: #6829
+
+; There are cases where Reassociation used to double-count a constant value
```

Comment:
I tried quite hard to reduce this test case, or to adapt the upstream test case.

Reassociation works iteratively, and it needs to generate certain intermediate results that then force the final issue.
It's very hard to follow. This is actually a cut down version of the original HLSL we got (and I put in the issue)